metadata-api/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/File.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.persistence.entities;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModelProperty;
+
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+@Entity
+@Table(uniqueConstraints = @UniqueConstraint(columnNames = {"accession", "version"}))
+public class File extends Auditable<Long> {
+
+    public enum Type {
+
+        tsv, tab, bam, bai, cram, crai, vcf, vcf_aggregate, bcf, tabix, wig, bed, gff, fasta, fastq, flatfile,
+        chromosome_list, sample_list, readme_file, phenotype_file, BioNano_native, Kallisto_native, agp,
+        unlocalised_list, info, manifest, other
```

Comment:
There are additional file types for "Run" files:
https://github.com/enasequence/schema/blob/master/src/main/resources/uk/ac/ebi/ena/sra/schema/SRA.run.xsd#L94-L320

Not sure would they ever be used, but just wanted to point it out.

Reply:
I am not which of the files we gonna use. the analysis file types may contain any old types as well .

Reply:
may be once the loading is ready , i could check the filetypes present

Reply:
> There are additional file types for "Run" files:
> https://github.com/enasequence/schema/blob/master/src/main/resources/uk/ac/ebi/ena/sra/schema/SRA.run.xsd#L94-L320
> 
> Not sure would they ever be used, but just wanted to point it out.

I took the filetypes from Analysis filetypes supported Enum class , so i think others wont be supported or would be tagged as OTHER